src/clients/c++/experimental_api_v2/library/common.h

```diff
@@ -0,0 +1,250 @@
+  /// indicate no error.
+  static const Error Success;
+
+ private:
+  friend std::ostream& operator<<(std::ostream&, const Error&);
+  std::string msg_;
+};
+
+//==============================================================================
+/// Structure to hold options for Inference Request.
+///
+struct InferOptions {
+  explicit InferOptions(const std::string& model_name)
+      : model_name_(model_name), model_version_(""), request_id_(""),
+        sequence_id_(0), sequence_start_(false), sequence_end_(false),
+        priority_(0), timeout_(0)
+  {
+  }
+  /// The name of the model to run inference.
+  std::string model_name_;
+  /// The version of the model to use while running inference. The default
+  /// value is an empty string which means the server will select the
+  /// version of the model based on its internal policy.
+  std::string model_version_;
+  /// An identifier for the request. If specified will be returned
+  /// in the response. Default value is an empty string which means no
+  /// request_id will be used.
+  std::string request_id_;
+  /// The unique identifier for the sequence being represented by the
+  /// object. Default value is 0 which means that the request does not
+  /// belong to a sequence.
+  uint64_t sequence_id_;
+  /// Indicates whether the request being added marks the start of the
+  /// sequence. Default value is False. This argument is ignored if
+  /// 'sequence_id' is 0.
+  bool sequence_start_;
+  /// Indicates whether the request being added marks the end of the
+  /// sequence. Default value is False. This argument is ignored if
+  /// 'sequence_id' is 0.
+  bool sequence_end_;
+  /// Indicates the priority of the request. Priority value zero
+  /// indicates that the default priority level should be used
+  /// (i.e. same behavior as not specifying the priority parameter).
+  /// Lower value priorities indicate higher priority levels. Thus
+  /// the highest priority level is indicated by setting the parameter
+  /// to 1, the next highest is 2, etc. If not provided, the server
+  /// will handle the request using default setting for the model.
+  uint64_t priority_;
+  /// The timeout value for the request, in microseconds. If the request
+  /// cannot be completed within the time the server can take a
+  /// model-specific action such as terminating the request. If not
+  /// provided, the server will handle the request using default setting
+  /// for the model.
+  uint64_t timeout_;
+};
+
+//==============================================================================
+/// An interface for InferInput object to describe the model input for
+/// inference.
+///
+class InferInput {
+ public:
+  /// Get the name of input associated with this object
+  /// \param name Returns the name of input.
+  /// \return Error object indicating success or failure of the
+  /// request.
+  virtual Error GetName(std::string* name) const = 0;
```

Comment:
Elsewhere we don't use "Get" style naming. We would just say Name() and SetName(). Can we follow that same convention in the C++ API.